PhpcrMigration/Application/Parser/PropertyNodeParser.php

```diff
@@ -144,15 +150,88 @@ private function getPropertyPath(string $propertyPath, string $name): string
         } elseif (\str_starts_with($name, 'excerpt-')) {
             $name = \substr($name, 8);
             $propertyPath .= '[excerpt][' . $name . ']';
-        } elseif (\preg_match('/^(.+)#(\d+)$/', $name, $matches)) {
-            $name = $matches[1];
-            $index = (int) $matches[2];
-            [$blocksKey, $type] = \explode('-', $name);
-            $propertyPath .= '[' . $blocksKey . '][' . $index . '][' . $type . ']';
+        } elseif (\str_contains($name, '#') || \str_ends_with($name, '-length')) {
+            $propertyPath .= $this->parseBlockPropertyPath($name);
         } else {
             $propertyPath .= '[' . $name . ']';
         }
 
         return $propertyPath;
     }
+
+    private function parseBlockPropertyPath(string $path): string
+    {
+        $segments = \explode('-', $path);
+        $result = '';
+
+        // First segment is always a block name
+        $result .= '[' . $segments[0] . ']';
+        $counter = \count($segments);
+
+        for ($i = 1; $i < $counter; ++$i) {
+            $segment = $segments[$i];
+
+            if (\preg_match('/^(.+)#(\d+)$/', $segment, $matches)) {
+                // For segments with index (like 'block#0')
+                $name = $matches[1];
+                $index = (int) $matches[2];
+
+                $result .= '[' . $index . '][' . $name . ']';
+            } else {
+                // Regular segments without index (like type / length)
+                $result .= '[' . $segment . ']';
+            }
+        }
+
+        return $result;
+    }
+
+    /**
+     * @param mixed[] $blocks
+     *
+     * @return mixed[]
+     */
+    private function trimBlocksToLengths(array $blocks): array
+    {
+        $maxLength = null;
+        $isImageMap = false;
+        foreach ($blocks as $index => $item) {
+            if ('length' === $index) {
+                /** @var int $maxLength */
+                $maxLength = $item;
+            }
+
+            if ('imageId' === $index) {
```

Comment:
Not sure if this could result in false positives regarding image_maps

we check at the bottom if it has a `length` property (e.g. block or image_map) and additionally if we have an `imageId`, when both checks are true, we think that we have an image_map with hotspots 🤔

Reply:
As discussed: in the worst case an empty array `hotspots` is added to the properties. This will be overwritten as soon as the page/article saved/published again, but other than that there should be no implications.

We also cannot only add the `hotspots` array if it is not empty, because then the sulu admin will crash for migrated image_maps that do not have any hotspots